Access Management update in Portfolio Management

Luyao-Zhang-1 opened the pull request.

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update access-management.png

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update access-management.png

Luyao-Zhang-1 committed:
Update access-requests.png

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Merge branch 'development' into lz-portfolio-management

Luyao-Zhang-1 committed:
fix link

hikmet-erd commented on content/en/docs/control-center/portfolios.md:
```diff
@@ -33,9 +33,9 @@ The **All Portfolios** tab lists all portfolios in your company. Here you can [m
+* **Portfolio Name** – This is the name of the portfolio. Clicking the name opens a pop-up window that shows the general information of the portfolio, such as the company name, privacy settings of the portfolio, number of active members (deactivated users are not counted), and a list of the Portfolio Managers.
 * **Managers** – This shows the number of Portfolio Managers in this portfolio.
-* **Members** – This shows the number of users who have access to this portfolio.
+* **Members** – This only shows the number of active users of this portfolio. The deactivated users are not included in this number.
```
'only' might be unnecessary here, is better to just say 'This shows'?

hikmet-erd commented on content/en/docs/control-center/portfolios.md:
```diff
@@ -59,24 +59,40 @@ If a portfolio has an alert icon (**⚠**), that means it no longer has a Portfo
-    * **Edit** – Selecting this allows you to change the role of the user.
-    * **Remove** – Selecting this allows you to remove the user from the portfolio.
+* Ellipsis button (**...**) – Clicking the button opens a menu with the following items:
+  * **Edit** – Selecting this allows you to change the role of the user.
```
Should be 'Edit Role'

hikmet-erd commented on content/en/docs/control-center/portfolios.md:
```diff
@@ -59,24 +59,40 @@ If a portfolio has an alert icon (**⚠**), that means it no longer has a Portfo
+After you click the ellipsis button (**...**) in the list on the [All Portfolios](#all-portfolios) tab or the [Alerts](#alerts) tab, you can choose **Access Management** to open the **Access Management** page. The Access Management page contains two tabs: the **Members** tab and the **Pending Invites** tab.
 
-In the search box, you can search for a user by the user name, email, company, status, or role.
+#### 2.1.1 Memebers
```
Typo at Members

hikmet-erd commented on content/en/docs/control-center/portfolios.md:
```diff
@@ -59,24 +59,40 @@ If a portfolio has an alert icon (**⚠**), that means it no longer has a Portfo
-In the search box, you can search for a user by the user name, email, company, status, or role.
+#### 2.1.1 Memebers
+
+On the **Member** tab, you can search for a user in the search box by the user name, email, company, status, or role.
```
Should be 'On the Members tab'

hikmet-erd replied:
We cannot search on status in this page.

hikmet-erd commented on content/en/docs/control-center/portfolios.md:
```diff
@@ -59,24 +59,40 @@ If a portfolio has an alert icon (**⚠**), that means it no longer has a Portfo
+
+#### 2.1.2 Pending Invites
+
+The **Pending Invites** tab shows all pending invites for the portfolio. You can search a user by their email. You can also filters users by their role. The list contains the following items:
```
Typo at 'filters', should be 'filter'

hikmet-erd replied:
Should we clarify here a bit what a Pending invite is? That it is for external users and they need to be accepted first by that user.

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -435,11 +435,32 @@ Alternatively, to delete an archived initiative, you can also click the ellipsis
-There are three access roles: **Portfolio Managers**, **Contributors**, and **Viewers**.
+### 5.1 Members
+
+The **Members** tab lists all the users who can access the portfolio. You can search a user by their user name, email, or company. You can also filters users by their role and status. The list contains the following items:
```
Typo at 'filters', should be 'filter'

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -465,49 +486,88 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
 The users that you invited receive a notification per email and now appear on the **Access Management** page. 
 
-For users outside your company, they need to accept their invitation. Once they accept, their **Status** will change from **Pending** to **Active**. 
+For users outside your company, they need to accept the invitation. They will appear on the **Pending Invites** tab until they accept the invitation. After they accept the invitation, they will appear in the **Members** tab.
```
They will appear on the **Pending Invites** tab until they accept or reject the invitation.

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -465,49 +486,88 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
 {{< figure src="/attachments/developerportal/portfolio-management/access-requests.png" >}}
 
-The **Access Requests** page shows all open access requests for the portfolio with the following items:
+The **Access Requests** tab shows all open access requests for the portfolio. You can search a user by their user name. You can also filters users by their role. The list contains the following items:
```
Typo at 'filters', should be 'filter'

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -465,49 +486,88 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
+
+* **Email** – This shows the email of the invited user.
+* **Role** – This shows the access role of the invited user.
+* **Invite date** – This shows the date the invite was sent.
```
Should be 'Invite Date' with capital

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -465,49 +486,88 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
+### 5.3 Pending Invites
 
 {{% alert color="info" %}}
+This tab is only visible to Project Managers.
```
Should be 'Portfolio Managers'

hikmet-erd commented on content/en/docs/control-center/portfolios.md:
```diff
@@ -59,24 +59,40 @@ If a portfolio has an alert icon (**⚠**), that means it no longer has a Portfo
 
 The **Add Users** button on the upper-right corner allows you to invite new users to the portfolio using their email addresses.
 
+The **Remove Deactivated Users** button allows you to remove all deactivated users from this portfolio. If a deactivated user that you will remove still owns initiatives, these initiatives will no longer have an owner after you remove the deactivated user from the portfolio.
```
Should we also mention this button is visible only if there is any deactivated users in the portfolio?

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -465,49 +486,88 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
+{{% /alert %}}
 
-The **Portfolio Settings** page is only available for Portfolio Managers.
+The **Pending Invites** tab shows all pending invites for the portfolio. You can search a user by their email. You can also filters users by their role. The list contains the following items:
```
Typo at 'filters', should be 'filter'

hikmet-erd replied:
Should we explain here what is a pending invite and how it happens?(due to external users being invited)

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -465,49 +486,88 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
+{{% /alert %}}
 
-The **Portfolio Settings** page is only available for Portfolio Managers.
+The **Pending Invites** tab shows all pending invites for the portfolio. You can search a user by their email. You can also filters users by their role. The list contains the following items:
```
Should we elaborate here a bit on what is a pending invite and how it is created?

Luyao-Zhang-1 replied:
I rewrote the text here as follows:

> When a Portfolio Manager invites a user outside the company to join the portfolio, the user gets an invite via email. The user must first accept the invite, then they can access this portfolio. Before the user accept or reject the invite, the invite appears on the **Pending Invites** tab. After they accept the invitation, they will appear on the **Members** tab.

hikmet-erd replied:
'Before' gives the impression of a moment, while 'until' is a period so do you think this is better alternative for this part?

'Before the user accept or reject the invite'
'Until the user accepts or rejects the invite'

hikmet-erd commented on content/en/docs/control-center/portfolios.md:
```diff
@@ -59,24 +59,40 @@ If a portfolio has an alert icon (**⚠**), that means it no longer has a Portfo
 
+The **Remove Deactivated Users** button allows you to remove all deactivated users from this portfolio. If a deactivated user that you will remove still owns initiatives, these initiatives will no longer have an owner after you remove the deactivated user from the portfolio.
+
 The list shows all users who have access to the portfolio as well as the pending portfolio invitations. In the list, you can see the following items:
```
Should not mention pending invitations here.

hikmet-erd replied:
Should we mention this sentence at the beginning of this section?

hikmet-erd commented on content/en/docs/control-center/portfolios.md:
```diff
@@ -33,9 +33,9 @@ The **All Portfolios** tab lists all portfolios in your company. Here you can [m
 
```
Duplication between lines 10-28

hikmet-erd commented on content/en/docs/control-center/portfolios.md:
```diff
@@ -59,24 +59,40 @@ If a portfolio has an alert icon (**⚠**), that means it no longer has a Portfo
+* **Invite date** – This shows the date the invite was sent.
+* Ellipsis button (**...**) – Clicking the button opens a menu with the following items:
+  * **Edit Role** – Selecting this allows you to change the access role of the pending invite.
+  * **Delete** – Selecting this allows you to cancel the pending invite.
```
Why not just say "delete" or "remove" instead of 'cancel'?

Luyao-Zhang-1 replied:
@hikmet-erd I think "Cancel" implies the consequence of this action more in a social context. When an invite is canceled, it is withdrawn and doesn't stand anymore.
"Delete" or "Remove" can just mean removing the invite from the database or in the UI.  What do you think?

hikmet-erd replied:
You have a point then we might change the caption of this button later to have consistency.

hikmet-erd commented on content/en/docs/control-center/portfolios.md:
```diff
@@ -96,11 +112,11 @@ The **Alerts** tab lists all the portfolios that no longer have a Portfolio Mana
 * **Managers** – This shows the number of Portfolio Managers in this portfolio.
 
-* **Members** – This shows the number of users who have access to the portfolio.
+* **Members** – This only shows the number of active users of this portfolio. The deactivated users are not included in this number.
```
'only' is not needed here

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -465,49 +486,88 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
+
+  * **Edit Role** – Selecting this allows you to change the access role of the pending invite.
+
+  * **Delete** – Selecting this allows you to cancel the pending invite.
```
Why not just say "delete" or "remove" instead of 'cancel'?

Luyao-Zhang-1 replied:
Please see above.

hikmet-erd replied:
We agree with your point

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -465,49 +486,88 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
+### 5.2 Access Requests {#access-requests}
+
+{{% alert color="info" %}}
+This tab is only visible to Project Managers.
```
Should be 'Portfolio Managers'

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -435,11 +435,32 @@ Alternatively, to delete an archived initiative, you can also click the ellipsis
+
+* Ellipsis button (only available to Portfolio Managers) – Clicking the button opens a menu with the following items:
+
+  * **Edit** – Selecting this allows you to change the role of the user.
```
Should be 'Edit Role'

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -465,49 +486,88 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
+
+{{% alert color="info" %}}Only [Portfolio Managers](#access-management) can edit or remove a user.{{% /alert %}}
 
 At the end of the row, click the ellipsis button (**...**), and then select the corresponding action. To update the permissions and the role of a user, click **Edit**. To remove a user from the portfolio, click **Remove**.
```
should be ', click **Edit Role**'

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -465,49 +486,88 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
+
+You can remove all the deactivated users from the portfolio in one go. 
+
+To do so, click **Remove Deactivated Users** above the list on the right side. If a deactivated user that you will remove still owns initiatives, these initiatives will no longer have an owner after you remove the deactivated user from the portfolio.
```
Should we mention that this button is only visible if there is any deactivated users?

hikmet-erd committed:
Update _index.md

Small changes to 3.1.2 and 3.1.3 and to the access rights table

hikmet-erd commented on static/attachments/developerportal/portfolio-management/access-management.png:
We can blur out the company names here as it is not a proper company name

Luyao-Zhang-1 committed:
Merge pull request #6918 from hikmet-erd/patch-2

Update _index.md

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update access-management.png

Luyao-Zhang-1 committed:
Delete edit-delete-user.png

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update _index.md

hikmet-erd commented on content/en/docs/control-center/portfolios.md:
```diff
@@ -59,24 +49,42 @@ If a portfolio has an alert icon (**⚠**), that means it no longer has a Portfo
+
+#### 2.1.2 Pending Invites
+
+When a Portfolio Manager invites a user outside the company to join the portfolio, the user gets an invite via email. The user must first accept the invite, then they can access this portfolio. Until the user accept or reject the invite, the invite appears on the **Pending Invites** tab. After they accept the invitation, they will appear on the **Members** tab.
```
'Until the user accept or reject', should this be 'user accepts or rejects'?

hikmet-erd replied:
Or, user has a plural usage as well?

Luyao-Zhang-1 replied:
Thanks for catching this. It should be `accepts` or `rejects`. I will fix it.

hikmet-erd commented on content/en/docs/control-center/portfolios.md:
```diff
@@ -59,24 +49,42 @@ If a portfolio has an alert icon (**⚠**), that means it no longer has a Portfo
+
+* **Email** – This shows the email of the invited user.
+* **Role** – This shows the access role of the invited user.
+* **Invite date** – This shows the date the invite was sent.
```
Date needs to be capital

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -464,50 +485,84 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
 
-### 5.1 Adding New Users {#add-users}
+* Delete the portfolio permanently together with all the data in the portfolio when you leave the portfolio; or
+* Assign another user to be the new Portfolio Manager before you leave the portfolio
```
Do we need a dot at the end?

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -464,50 +485,84 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
-For users outside your company, they need to accept their invitation. Once they accept, their **Status** will change from **Pending** to **Active**. 
+For users outside your company, they need to accept the invitation. They will appear on the **Pending Invites** tab until they accept or reject the invitation. After they accept the invitation, they will appear in the **Members** tab.
+
+For users within your company, they do not need to accept their invitation. If the users have a Mendix  account, they will immediately be given access and appear on the **Members** tab. If they do not have an Mendix account, they will appear on the **Pending Invites** tab. After they create a Mendix account and log in to Portfolio Management, they will immediately be given access and appear on th ...
```
Double space between 'Mendix  account'

hikmet-erd replied:
'For users within your company, they do not need to accept their invitation. If the users have a Mendix  account' here

hikmet-erd commented on content/en/docs/developerportal/portfolio-management/_index.md:
```diff
@@ -464,50 +485,84 @@ The table below shows the permissions of Portfolio Managers, Contributors, and V
+{{% /alert %}}
 
-The **Portfolio Settings** page is only available for Portfolio Managers.
+When a Portfolio Manager invites a user outside the company to join the portfolio, the user gets an invite via email. The user must first accept the invite, then they can access this portfolio. Until the user accept or reject the invite, the invite appears on the **Pending Invites** tab. After they accept the invitation, they will appear on the **Members** tab.
```
'accepts or rejects' vs 'accept or reject' ?

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Remove one extra space

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Update latest-releases.html

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
add anchors

Luyao-Zhang-1 committed:
Update _index.md

hikmet-erd committed:
Update _index.md

'The Members tab lists all the users who can access the portfolio.' 

'The Members tab lists all the users that have portfolio access.'

Here we also list deactivated members who has access but cannot actually access the portfolio. I have reworded it to improve the meaning although slightly. 


I have added details on how to assign a portfolio manager. We had an internal feedback from another team about this part being ambigious.

Luyao-Zhang-1 committed:
remove extra Control Center instances

Luyao-Zhang-1 committed:
Merge pull request #6921 from hikmet-erd/patch-3

Update _index.md

Luyao-Zhang-1 committed:
Update portfolios.md

Luyao-Zhang-1 committed:
Update _index.md

Luyao-Zhang-1 committed:
Merge branch 'lz-portfolio-management' of https://github.com/mendix/docs into lz-portfolio-management

Luyao-Zhang-1 committed:
replace links

Luyao-Zhang-1 committed:
fix links
